patterns/README.md

```diff
@@ -1395,3 +1414,250 @@ substantial limitation when creating value labels or categoricals, since
+
+### Principles
+
+Two aspects need to be addressed:
+
+* **relationship expression**:
+
+   This methodology applied for databases can also be applied for tabular data whose structure is similar to that of relational database tables but whose representation of relationships is different (see [patterns](https://www.ietf.org/archive/id/draft-thomy-ntv-tab-00.html#section-2) used in tabular representations).
+
+   This variation is explained in the [linked notebook](https://github.com/loco-philippe/Environmental-Sensing/blob/main/property_relationship/methodology.ipynb) and presented in the [example](https://nbviewer.org/github/loco-philippe/Environmental-Sensing/blob/main/property_relationship/example_schema.ipynb).
+
+   Using a data model is a simple way to express relationships but it is not required. We can express the relationships directly at the data schema level.
+
+* **validity of a dataset**:
+
+   Checking the validity of a relationship for a defined dataset is one of the functions of [tabular structure analysis](https://github.com/loco-philippe/tab-analysis/blob/main/docs/tabular_analysis.pdf). It only requires counting functions accessible for any type of language (see [example of implementation](https://github.com/loco-philippe/Environmental-Sensing/blob/main/property_relationship/example.ipynb)).
+
+### Proposed extensions
+
+A relationship is defined by the following information:
+
+* the two Fields involved (the order of the Fields is important with the "derived" link),
+* the textual representation of the relationship,
+* the nature of the relationship
+
+Three proposals for extending Table Schema are presented:
+
+* **New Field Descriptor**:
+
+  A `relationships` Field Descriptor is added.
+  The properties associated with this Descriptor could be:
+
+  * `parent`: name of the other Field involved
+  * `description`: description string (optional)
+  * `link`: nature of the relationship
+
+  Pros:
+
+  * No mixing with other descriptors
+  * Consistent with a field view
+  * the direction of the relationship is clear
+
+  Cons:
+
+  * a relationship is not a property of a Field
+
+  Example:
+
+  ```json
+  { "fields" : [
+      { "name": "country",
+        "relationships": [
+          { "parent" : "code",
+            "description" : "is the country code alpha-2 of",
+            "link" : "coupled" }
+        ]
+      }
+      { "name": "region",
+        "relationships": [
+          { "parent" : "population",
+            "description" : "is the population of",
+            "link" : "derived"}
+        ]
+      }
+    ]
+  }
+  ```
+
+* **New Constraint Property**:
+
+  A `relationships` Property of `constraint` Descriptor is added.
+
+  The properties associated with this Property could be:
+
+  * `parent`: name of the other Field involved
+  * `description`: description string (optional)
+  * `link`: nature of the relationship
+
+  Pros:
+
+  * The `constraints` Descriptor is consistent with the point
```

Comment:
What do you mean with "with the point"?

Reply:
Your are right, it's not very clear ! I propose to replace it by "with the purpose of the proposal"

Reply:
commit done